packages/dashboard/test/DashboardServer.test.ts

```diff
@@ -0,0 +1,144 @@
+import Ganache from "ganache-core";
+import { providers, utils } from "ethers";
+import Web3 from "web3";
+import { getMessageBusPorts } from "@truffle/dashboard-message-bus";
+import MockDashboard from "./MockDashboard";
+import { DashboardServer } from "../lib";
+
+jest.setTimeout(200000);
+
+// TODO: These tests were copy-pasted from the browser-provider tests
+// We should figure out whether we want to make this DRYer
+describe("DashboardServer", () => {
```

Comment:
Is there a good way to add automated tests for reverted transactions and/or rejected requests? Maybe I'm missing these, but I didn't see them when I took a look just now.

Reply:
We don't have any reverted tx tests. Probably what we'd want to do is add some contract deployments / interactions to the automated testing. And then with those contract interactions it should be easy to include a couple of interactions that we expect to revert.